### PR TITLE
add details into internal validation

### PR DESCRIPTION
**Pull request by imanushin**
Initial purpose to change - add details into [YamlConfig](detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/YamlConfig.kt) validation.

And in addition, more details were added into the project internal error raise.

Please check [DetektGenerateConfigTask](detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt), because semantic was changed: from now actual error is provided to gradle. It means, that ```--stacktrace``` gradle option is able to show ```detekt```'s stack.

### CODE REVIEW

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/internal/Validation.kt [hunk 1]
```diff
@@ -6,5 +6,6 @@ private val identifierRegex = Regex("[aA-zZ]+([-][aA-zZ]+)*")
  * Checks if given string matches the criteria of an id - [aA-zZ]+([-][aA-zZ]+)* .
  */
 internal fun validateIdentifier(id: String) {
-    require(id.matches(identifierRegex)) { "id must match [aA-zZ]+([-][aA-zZ]+)*" }
+
+    require(id.matches(identifierRegex)) { "id must match ${identifierRegex.pattern}" }
```
Comment (BraisGabin) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/internal/Validation.kt]: Add the actual value

Comment (imanushin) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/internal/Validation.kt]: Will do, thanks

**Comment by imanushin** [comment]
Thanks!

I'll fix this at the next weekend. A little busy we weekdays, sorry.

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/internal/Validation.kt (hunk 1 above)
Comment (imanushin) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/internal/Validation.kt]: Fixed

File: detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/KtCompilerTest.kt [hunk 2]
```diff
@@ -26,7 +26,7 @@ class KtCompilerTest : Spek({
         it("throws an exception for an invalid sub path") {
             assertThatIllegalArgumentException()
                 .isThrownBy { ktCompiler.compile(path, path) }
-                .withMessage("Given sub path should be a regular file!")
+                .withMessage("Given sub path (/home/travis/build/arturbosch/detekt/detekt-core/build/resources/test/cases) should be a regular file!")
```
Comment (BraisGabin) [inline detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/KtCompilerTest.kt]: I'm sure that this test is not going to pass on my computer. Is it possible to mock the path to a constant?

Comment (imanushin) [reply detekt-core/src/test/kotlin/io/gitlab/arturbosch/detekt/core/KtCompilerTest.kt]: @BraisGabin , definitely, thank you. Will fix.

**Comment by imanushin** [comment]
@BraisGabin, fixed. Apologise for that.

**Pull request merged**